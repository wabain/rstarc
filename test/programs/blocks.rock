
Fernpl takes X and Y
Give back X


While true is true
Until false is false
If true is true
Say "yeah"
Else
Say "nah"
