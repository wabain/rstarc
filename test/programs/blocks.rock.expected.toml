[tests]
enabled = [ "tokens", "pretty",]

[tokens]
returncode = "0"
stdout = "0..1 ⏎\n1..7 Fernpl\n8..13 takes\n14..15 X\n16..19 and\n20..21 Y\n21..22 ⏎\n22..26 give\n27..31 back\n32..33 X\n33..34 ⏎\n34..35 ⏎\n35..36 ⏎\n36..41 while\n42..46 true\n47..49 is\n50..54 true\n54..55 ⏎\n55..60 until\n61..66 false\n67..69 is\n70..75 false\n75..76 ⏎\n76..78 if\n79..83 true\n84..86 is\n87..91 true\n91..92 ⏎\n92..95 say\n96..102 \"yeah\"\n102..103 ⏎\n103..107 else\n107..108 ⏎\n108..111 say\n112..117 \"nah\"\n117..118 ⏎\n118..118 ⏎\n118..118 ⏎\n118..118 ⏎\n118..118 ⏎\n118..118 <eof>\n"

[pretty]
returncode = "0"
stdout = "Fernpl takes X and Y\nGive back X\n\nWhile true is true\nUntil false is false\nIf true is true\nSay \"yeah\"\n\nElse\nSay \"nah\"\n\n\n\n"
